cprover-cpp-api/src/api.h

```diff
@@ -0,0 +1,38 @@
+// Author: Fotis Koutoulakis for Diffblue Ltd.
+
+#ifndef CPROVER_API_API_H
+#define CPROVER_API_API_H
+
+#include <memory>
+#include <string>
+#include <vector>
+
+// Forward declaration of API dependencies
+class goto_modelt;
+class message_handlert;
+class optionst;
+
+#include "options.h"
+
+// An object in the pattern of Session Facade - owning all of the memory
+// the API is using and being responsible for the management of that.
```

Comment:
When is this memory released? When the object drops?

Reply:
Yes - this owns unique pointers to all of its dependencies, so as soon as the object gets out of scope all of the memory the object uses should be released.